### ISSUE TITLE
Incremental Proof Generation

### ISSUE THREAD

**Issue by crisgarner** [privacy-scaling-explorations/maci#1895]
# Details

Deleting proofs from the proof directory will regenerate proofs resulting on issues when adding the tally as commitments will be different due to the use of random salts. 

The idea is to use the --incremental flag to only allow the regeneration if the user inputs the flag.

**Comment by 0xmad** [comment]
@JohnGuilding yes, it's still relevant. Here are more details for this task (do not take a look at code, it's probably AI generated and doesn't fix the issue)
https://github.com/privacy-scaling-explorations/maci/pull/2337.
